lib/Doctrine/ODM/PHPCR/Tools/Console/Command/DocumentConvertTranslationCommand.php

```diff
@@ -0,0 +1,109 @@
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * This software consists of voluntary contributions made by many individuals
+ * and is licensed under the MIT license. For more information, see
+ * <http://www.doctrine-project.org>.
+ */
+
+namespace Doctrine\ODM\PHPCR\Tools\Console\Command;
+
+use Doctrine\ODM\PHPCR\Tools\Helper\TranslationConverter;
+use PHPCR\SessionInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author David Buchmann <mail@davidbu.ch>
+ */
+class DocumentConvertTranslationCommand extends Command
+{
+    private $translationConverter;
+
+    public function __construct($name = null, TranslationConverter $translationConverter = null)
+    {
+        parent::__construct($name);
+        $this->translationConverter = $translationConverter;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('doctrine:phpcr:document:convert-translation')
+            ->setDescription('Convert fields to translated or back to untranslated, and between different strategies after a refactoring.')
+
+            ->addArgument('classname', InputArgument::REQUIRED, 'Class that has changed translation information')
+            ->addOption('previous-strategy', 'prev', InputOption::VALUE_OPTIONAL,
+                'Name of the previous translation strategy if there was one. Omit for converting from non-translated to translated'
+            )
+            ->addOption('fields', null, InputOption::VALUE_IS_ARRAY|InputOption::VALUE_OPTIONAL,
+                'The fields to convert. If not specified, all fields configured as translated will be converted.',
+                array()
+            )
+            ->setHelp(<<<HERE
+The <info>doctrine:phpcr:docment:convert-translation</info> command migrates translations
+from a previous format to the current mapping.
+
+  <info>$ php ./app/console/phpcr doctrine:phpcr:document:convert-translation "Document\\ClassName"</info>
+
+<comment>When some fields already where translated, you need to specify which fields to convert.</comment>
+Failing to do that would erase all fields already translated previously.
+
+Note that when only some fields changed or when converting between translation strategies, you need
+to specify the previous strategy. When converting to untranslated, you additionally need to specify
+the fields that previously where translated.
+HERE
+            );
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        /** @var $session SessionInterface */
+        $session = $this->getHelper('phpcr')->getSession();
```

Comment:
that sounds like a good idea. and showing what will happen would be great.

i can add a --force parameter and otherwise ask for confirmation. for
the rest, i don't see that i have time to do that soon enough. i don't
want to delay the release. the --force is a BC break, providing more
information is possible later as well. we would need to refactor the
class that does the migration to be able to explain its decisions
without executing them yet.